mediachain/datastore/ipfs.py

```diff
@@ -0,0 +1,30 @@
+import ipfsApi
+import cbor
+from tempfile import NamedTemporaryFile
+from mediachain.datastore.data_objects import Record, MultihashReference
+
+
+class IpfsDatastore(object):
+    def __init__(self, host='127.0.0.1', port=5001):
+        self.client = ipfsApi.Client(host, port)
+
+    def put(self, data_object):
+        if isinstance(data_object, Record):
+            content = data_object.to_cbor_bytes()
+        else:
+            content = data_object
+
+        with NamedTemporaryFile() as f:
+            f.write(content)
+            f.flush()
+            result = self.client.add(f.name)
+        return result[1][u'Hash']
```

Comment:
I was getting naked unicode strings rather than arrays when adding direct paths, but I guess if adding something in `/tmp/...` it will actually add both so you'll get an array. You probably want `[-1]` though because tmp path is not guaranteed to have depth = 1